### PR TITLE
Adding basic controller test without need of VIDEO driver

### PR DESCRIPTION
**Pull request by fjtrujy**
## Description
This PR basically adds a new test sample for checking in the implementation of Joystick is working properly.
The existing examples required to have `VIDEO DRIVER` as well, however, the one that I'm adding doesn't require it at all.

Cheers

### CODE REVIEW

File: test/basiccontrollermap.c [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        if (event.type == SDL_QUIT) {
+            break;
+        } else if (event.type == SDL_KEYDOWN) {
+            if (event.key.keysym.sym == SDLK_ESCAPE) {
+                break;
+            }
+        } else if (event.type == SDL_JOYAXISMOTION) {
+            SDL_Log("Axis: %i %i\n", event.jaxis.axis, event.jaxis.value);
+        } else if (event.type == SDL_JOYBUTTONDOWN) {
+            SDL_Log("Button: %i\n", event.jbutton.button);
```
Comment (madebr) [inline test/basiccontrollermap.c:100]: I think this should be a switch.
Also, the other controller example handles `SDL_JOYDEVICEREMOVED`

File: test/basiccontrollermap.c [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+
+int main(int argc, char *argv[]) {
+    SDL_Event event;
+    SDL_Joystick *joy;
+    const char *type;
+    char guid[64];
+
+    SDL_Init(SDL_INIT_JOYSTICK);
+    joy = SDL_JoystickOpen(0);
```
Comment (madebr) [inline test/basiccontrollermap.c:33]: I don't want to be "a joy breaker", but `joy` can be `NULL` :rofl:  (I'll show myself out)

The other examples test for `NULL` and log it appropriately.

File: test/basiccontrollermap.c (hunk 1 above)
Comment (fjtrujy) [reply test/basiccontrollermap.c]: Applied!

File: test/basiccontrollermap.c (hunk 2 above)
Comment (fjtrujy) [reply test/basiccontrollermap.c:34]: Thanks! Done!

File: test/basiccontrollermap.c [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    
+    SDL_Delay(1000);
+    SDL_Log("Checking Rumble functionality for 2 seconds\n");
+    SDL_JoystickRumble(joy, 0xAAFF, 0xAAFF, 2000);
+
+    while (running) {
+        SDL_WaitEvent(&event);
+        switch(event.type) {
+            case SDL_QUIT:
+                break;
```
Comment (madebr) [inline test/basiccontrollermap.c:99]: Is the `SDL_QUIT` event type ever emitted for a console application?
Also, `running` should be set to false here.

File: test/basiccontrollermap.c [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+                    running = 0;
+                break;
+            case SDL_JOYAXISMOTION:
+                SDL_Log("Axis: %i %i\n", event.jaxis.axis, event.jaxis.value);
+                break;
+            case SDL_JOYBUTTONDOWN:
+                SDL_Log("Button: %i\n", event.jbutton.button);
+                break;
+            case SDL_JOYDEVICEREMOVED:
+                SDL_Log("Joy removed: %i\n", event.jdevice.which);
```
Comment (madebr) [inline test/basiccontrollermap.c:111]: The event loop should also quit when, for the removed device, `event.jdevice.which == SDL_JoystickInstanceID(joy)`.
(`testjoystick.c` does this)

File: test/basiccontrollermap.c (hunk 3 above)
Comment (fjtrujy) [reply test/basiccontrollermap.c:99]: I don't know if `SDL_QUIT` could be emitted for a console ....
I have applied the running to false here.

File: test/basiccontrollermap.c (hunk 4 above)
Comment (fjtrujy) [reply test/basiccontrollermap.c:111]: Applied!

**Comment by slouken** [comment]
Windows and macOS require the video driver and event loops for controller support.

What’s the motivation for removing video initialization?

**Comment by fjtrujy** [comment]
> Windows and macOS require the video driver and event loops for controller support.
> 
> What’s the motivation for removing video initialization?

I was implementing the Joystick driver and with the existing test examples was
Imposible to use them for checking the implementation, because they required the video driver, which I didn’t implement it yet.

**Comment by slouken** [comment]
In that case let's leave this un-merged and you can keep it in your branch until the video driver is implemented.

**Comment by fjtrujy** [comment]
> In that case let's leave this un-merged and you can keep it in your branch until the video driver is implemented.

But sorry I don’t understand it then…..
The main purpose of this example is to have a test controller example that doesn’t require any additional driver.
We have in the project already some controller examples that requires the usage of the video driver.
If you want to add here the video driver, then this specific test is useless, because there are similar tests in the project already.

If you don’t see the use case of this PR, you can safely close it.

Thanks

**Comment by slouken** [comment]
I understand your need for this test, but it won’t work on other platforms, so isn’t a good example and test for the functionality broadly available in SDL.

**Comment by ccawley2011** [comment]
> I was implementing the Joystick driver and with the existing test examples was Imposible to use them for checking the implementation, because they required the video driver, which I didn’t implement it yet.

I think in this case, it would be simpler to use the dummy video driver with the PS2 port until a proper one is implemented. It can be enabled by setting `SDL_VIDEODRIVER=dummy` in the environment or by calling `SDL_SetHint(SDL_HINT_VIDEODRIVER , "dummy")` before `SDL_Init()` is called.

**Pull request closed without merging**